mlir/lib/Dialect/Rock/utility/loweringUtils.cpp

```diff
@@ -344,3 +352,81 @@ FailureOr<RegsAsMatrixSubTiles> mlir::rock::getPackedRegsAsTileViews(
   }
   return gpuViews;
 }
+
+TopDownTMBuilder mlir::rock::swapThreadIdAndIteration(
+    TopDownTMBuilder &toMatrixC, ArrayRef<int64_t> bidGridLengths,
```

Comment:
bidGridLengths should not be needed here. I mean how the grid is partitioned should not be relevant.
I think what you need is mblocks and nblocks.

Reply:
to clarify further
bidGridLengths[1] == mblocks and  bidGridLengths[2] == nblocks will not always be the case..